src/Services/StatePersisterWrapper.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Pinnacle\OpenIdConnect\Services;
+
+use GuzzleHttp\Psr7\Uri;
+use Pinnacle\OpenIdConnect\Exceptions\StatePersisterMissingValueException;
+use Pinnacle\OpenIdConnect\Models\Constants\StateKey;
+use Pinnacle\OpenIdConnect\Models\Contracts\ProviderConfigurationInterface;
+use Pinnacle\OpenIdConnect\Models\Contracts\StatePersisterInterface;
+use Pinnacle\OpenIdConnect\Models\ProviderConfiguration;
+
+class StatePersisterWrapper
+{
+    public function __construct(private StatePersisterInterface $statePersister, private string $stateKey)
+    {
+    }
+
+    public function storeChallenge(string $challenge): void
```

Comment:
It seems like we're going to have to validate values all over the place. I think it might be better to use some value objects in place of a lot of the primitive string types in this library.

Reply:
You are right. I was getting that feeling as well while building this out. I should have followed my gut.